backend/src/main/java/com/bakdata/conquery/models/query/statistics/BalancingHistogram.java

```diff
@@ -0,0 +1,268 @@
+package com.bakdata.conquery.models.query.statistics;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.List;
+import java.util.Objects;
+
+import groovy.lang.Tuple;
+import groovy.lang.Tuple2;
+import it.unimi.dsi.fastutil.doubles.DoubleArrayList;
+import it.unimi.dsi.fastutil.doubles.DoubleIterator;
+import it.unimi.dsi.fastutil.doubles.DoubleList;
+import lombok.Data;
+import lombok.ToString;
+
+@Data
+public class BalancingHistogram {
+	private final Node[] nodes;
+	private final double min, max;
+	private final double width;
+
+	private final int expectedBins;
+
+
+	private int total;
+
+	public static BalancingHistogram create(double min, double max, int expectedBins) {
+		final double width = (max - min) / expectedBins;
+
+		return new BalancingHistogram(new Node[expectedBins], min, max, width, expectedBins);
+	}
+
+	public void add(double value) {
+		total++;
+
+		final int index;
+
+		if (value >= max) {
+			index = nodes.length - 1;
+		}
+		else if (value <= min) {
+			index = 0;
+		}
+		else {
+			index = (int) Math.floor((value - min) / width);
+		}
+
+		if (nodes[index] == null) {
+			nodes[index] = new Node(new DoubleArrayList());
+		}
+
+		nodes[index].add(value);
+	}
+
+	public List<Node> balanced(double stiffness) {
```

Comment:
Doku zu stiffness bitte. was bedeutet ein hoher/niedriger Wert. In welcher Range ist der Wert

Reply:
Der Code kommt wieder weg, aber theoretisch war es dafür da zu große/kleine bins zu splitten/mergen. Das verhindert zwar, dass bins extreme ausreise sind, aber ist schwer zu lesen.

Reply:
Mit stiffness hat man 0-1 gesteuert, wie groß sie von der expetcedBinSize abweichen dürfen.